src/puppetlabs/trapperkeeper/services/authorization/authorization_service.clj

```diff
@@ -0,0 +1,24 @@
+(ns puppetlabs.trapperkeeper.services.authorization.authorization-service
+  (:require [puppetlabs.trapperkeeper.core :refer [defservice]]
+            [puppetlabs.trapperkeeper.services :refer [service-context]]
+            [puppetlabs.trapperkeeper.authorization.ring-middleware :refer
+             [wrap-authorization-check]]
+            [puppetlabs.trapperkeeper.services.authorization.authorization-core :refer :all]))
+
+(defprotocol AuthorizationService
+  (wrap-handler [this handler]))
+
+(defservice authorization-service
+  AuthorizationService
+  [[:ConfigService get-in-config]]
+  (init
+    [this context]
+    (let [config (get-in-config [:authorization :rules])]
+      (validate-auth-config! config)
+      (assoc-in context [:authorization :rules] (transform-config config))))
```

Comment:
Not sure we need to nest the transformed config under ":authorization" since the context is specific to this service anyway, which is all about authorization.  Could just put it under ":rules" at the top-level.